### ISSUE TITLE
Warning on `matches!` with rule `redundant_pattern_matching`?

### ISSUE THREAD

**Issue by fanatid** [rust-lang/rust-clippy#6459]
Currently, clippy produce a warning on `redundant_pattern_matching` rule for constructions like:
```rust
if let Poll::Ready(_) = Poll::Ready(42) {}
```
but do not produce if we use `matches!`:
```rust
if matches!(Poll::Ready(42), Poll::Ready(_)) {}
```

I think that warning also should be produced for `matches!`, i.e. only `if Poll::Ready(42).is_ready() {}` should be valid (same for `Result`, `Option`, `std::net::IpAddr`).

**Comment by torfsen** [comment]
I think this has been partially fixed in the meantime, at least for `Option`:
```
error: redundant pattern matching, consider using `is_some()`
  --> tests/ui/redundant_pattern_matching_if_let_true.rs:42:8
   |
LL |     if matches!(Some(42), Some(_)) {}
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `Some(42).is_some()`
```
But as far as I can tell, the `matches!` expression from the OP still isn't caught.
